src/webapi/selection.rs

```diff
@@ -0,0 +1,341 @@
+    ///
+    /// Gecko browsers allow multiple selections across table cells.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/rangeCount)
+    pub fn range_count(&self) -> u32 {
+        js! (
+            return @{self}.rangeCount;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the type of the current selection.
+    ///
+    /// Possible values are:
+    ///
+    /// * `None`: No selection has currently been made.
+    /// * `Caret`: The selection is collapsed (i.e. the caret is placed on some text, but no
+    /// range has been selected).
+    /// * `Range`: A range has been selected.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/type)
+    pub fn selection_type(&self) -> String {
+        js! (
+            return @{self}.type;
+        ).try_into().unwrap()
+    }
+
+    /// Returns a [Range](struct.Range.html) object representing one of the ranges currently
+    /// selected.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/getRangeAt)
+    pub fn get_range_at(&self, index: u32) -> Range {
+        js! (
+            return @{self}.getRangeAt(@{index});
+        ).try_into().unwrap()
+    }
+
+    /// Adds a [Range](struct.Range.html) to the [Selection](struct.Selection.html).
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/addRange)
+    pub fn add_range(&self, range: Range) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.addRange(@{range});
+        };
+        Ok(())
+    }
+
+    /// Removes a [Range](struct.Range.html) from the [Selection](struct.Selection.html).
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/removeRange)
+    pub fn remove_range(&self, range: Range) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.removeRange(@{range});
+        };
+        Ok(())
+    }
+
+    /// Removes all ranges from the [Selection](struct.Selection.html), leaving the
+    /// [anchor_node](struct.Selection.html#method.anchor_node) and
+    /// [focus_node](struct.Selection.html#method.focus_node) properties equal to
+    /// `None` and leaving nothing selected.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/removeRange)
+    pub fn remove_all_ranges(&self) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.removeAllRanges();
+        };
+        Ok(())
+    }
+
+    /// Collapses the [Selection](struct.Selection.html) to a single point. The document is not
+    /// modified. If the content is focused or editable, the caret will blink there.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapse)
+    pub fn collapse(&self, node: Node, offset: Option<u32>) -> Result<(), TODO> {
```

Comment:
1. This can throw `IndexSizeError`.
2. Accept any `INode` here. (`pub fn collapse< T: INode >( &self, node: &T, ... )`)